docs/pages/enterprise/workflow/resource-requests.mdx

```diff
@@ -0,0 +1,365 @@
+<Tile icon="cloud" title="Teleport Cloud" href="./resource-requests.mdx/?scope=cloud">
+</Tile>
+<Tile icon="building" title="Teleport Enterprise" href="./resource-requests.mdx/?scope=enterprise">
+</Tile>
+</TileSet>
+
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud", "enterprise"]}>
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+(!docs/pages/includes/tctl.mdx!)
+
+## Step 1/8. Create the requester role
+
+This role allows the requester to search for resources accessible by the
+`access` role (all resources by default) and request access to them.
+
+```yaml
+# requester.yaml
+kind: role
+version: v5
+metadata:
+  name: requester
+spec:
+  allow:
+    request:
+      search_as_roles:
+        - access
+```
+
+```code
+$ tctl create requester.yaml
+```
+
+## Step 2/8. Create the reviewer role
+
+This role allows the reviewer to approve all requests for the `access` role.
+
+```yaml
+# reviewer.yaml
+kind: role
+version: v5
+metadata:
+  name: reviewer
+spec:
+  allow:
+    review_requests:
+      roles:
+        - access
+```
+
+```code
+$ tctl create reviewer.yaml
+```
+
+## Step 3/8. Grant the roles to users
+
+Grant the `requester` and `reviewer` roles to existing users, or create new
+users to test this feature.
+Make sure the requester has a valid `login` so that they can view and access SSH
+nodes.
+
+```code
+$ tctl users add alice --roles requester --logins alice
+$ tctl users add bob --roles reviewer
+```
+
+For the rest of the guide we will assume that the `requester` role has been
+granted to a user named `alice` and the `reviewer` role has been granted to a
+user named `bob`.
+
+## Step 4/8. Search for resources
+
+First, log in as `alice`.
+
+```code
+$ tsh login --proxy teleport.example.com --user alice
+```
+
+Notice that `tsh ls` returns an empty list, because `alice` does not have access to any resources by default.
+```code
+$ tsh ls
+Node Name Address Labels
+--------- ------- ------
+```
+
+Then try searching for all available ssh nodes.
+
+```code
+$ tsh request search --kind node
+Name                                 Hostname    Labels       Resource ID
+------------------------------------ ----------- ------------ ------------------------------------------------------
+b1168402-9340-421a-a344-af66a6675738 iot         test=test    /teleport.example.com/node/b1168402-9340-421a-a344-af66a6675738
+bbb56211-7b54-4f9e-bee9-b68ea156be5f node        test=test    /teleport.example.com/node/bbb56211-7b54-4f9e-bee9-b68ea156be5f
+ 
+To request access to these resources, run
+> tsh request create --resource /teleport.example.com/node/b1168402-9340-421a-a344-af66a6675738 --resource /teleport.example.com/node/bbb56211-7b54-4f9e-bee9-b68ea156be5f \
+    --reason <request reason>
+```
+
+You can search for resources of kind `node`, `kube_cluster`, `db`, `app`, and `windows_desktop`.
+Advanced filters and queries are supported, see our
+[filtering reference](/setup/reference/cli.mdx/#resource-filtering).
```

Comment:
sorry my suggestion was a wrong suggestion, you need to do this for it to work (should work, i'd double check):

`../../setup/reference/cli.mdx#resource-filtering`

the `../..` takes you back to `docs/pages`

I had to do it this way when i wrote the predicate guide (look at url) [docs/pages/setup/reference/predicate-language.mdx](https://raw.githubusercontent.com/gravitational/teleport/master/docs/pages/setup/reference/predicate-language.mdx) and had to refer to files in `access-controls/guides/` which was outside of my current directory `setup/reference/`

```
e.g:
- [Dynamic Impersonation](../../access-controls/guides/impersonation.mdx#filter-fields)
- [RBAC for sessions](../../access-controls/reference.mdx#filter-fields)
```

not sure if there is another way of doing it :woman_shrugging: